### PR TITLE
Fix first dropdown selection ignored with custom value

### LINKED ISSUE #3115: [0.55RC2] Switching language in UI doesn't work
As a first step, I tried switching to French in the UI.

image

While I could select the option in the list, clicking on the validation button didn't do anything (no feedback at all). Even restarting the game didn't register the change.

### PR DESCRIPTION
**Pull request by slipher**
Fix #3115

### CODE REVIEW

File: src/cgame/rocket/rocketFormControlSelect.h [hunk 1]
```diff
@@ -119,22 +117,24 @@ class CvarElementFormControlSelect : public Rml::ElementFormControlSelect, publi
...
 		{
 			Rml::Element* e = GetOption(i);
 			Rml::String value = e->GetAttribute<Rml::String>( "value", Rml::String() );
 			if ( value == cvarValue )
 			{
-				// If the selection is already correct, do not force set the cvar.
-				ignore_value_change = GetSelection() != i;
+				ignore_value_change = true;
 				SetSelection( i );
+				ignore_value_change = false;
```
Comment (DolceTriade) [inline src/cgame/rocket/rocketFormControlSelect.h:129]: why we need to do this? Does setselection trigger an event? maybe comment this?

File: src/cgame/rocket/rocketFormControlSelect.h [hunk 2]
```diff
@@ -119,22 +117,24 @@ class CvarElementFormControlSelect : public Rml::ElementFormControlSelect, publi
...
 				return;
 			}
 		}
 		// No matches...Let's add an option for our current value.
 		Add( Str::Format( "Custom Value (%s)", cvarValue ), cvarValue, 0, false );
-		SetSelection( 0 );
 		ignore_value_change = true;
+		SetSelection( 0 );
+		ignore_value_change = false;
```
Comment (DolceTriade) [inline src/cgame/rocket/rocketFormControlSelect.h:137]: Sounds like this should be a helper function if we always have to do this

File: src/cgame/rocket/rocketFormControlSelect.h (hunk 1 above)
Comment (slipher) [reply src/cgame/rocket/rocketFormControlSelect.h:129]: Added comment on ignore_value_change.

File: src/cgame/rocket/rocketFormControlSelect.h (hunk 2 above)
Comment (slipher) [reply src/cgame/rocket/rocketFormControlSelect.h:137]: Eh, we only do it twice, I like the rule of three :)

**Review by DolceTriade — APPROVED** [review]

**Pull request merged**